cram/cram_io.c

```diff
@@ -1893,15 +1893,29 @@ static int cram_populate_ref(cram_fd *fd, int id, ref_entry *r) {
     char *ref_path = getenv("REF_PATH");
     SAM_hdr_type *ty;
     SAM_hdr_tag *tag;
-    char path[PATH_MAX], path_tmp[PATH_MAX];
+    char path[PATH_MAX], path_tmp[PATH_MAX], cache[PATH_MAX];
     char *local_cache = getenv("REF_CACHE");
     mFILE *mf;
 
     if (fd->verbose)
 	fprintf(stderr, "cram_populate_ref on fd %p, id %d\n", fd, id);
 
-    if (!ref_path || *ref_path == 0)
+    if (!ref_path || *ref_path == 0) {
+	/*
+	 * If we have no ref path, we use the EBI server.
+	 * However to avoid spamming it we require a local ref cache too.
+	 */
 	ref_path = "http://www.ebi.ac.uk:80/ena/cram/md5/%s";
+	if (!local_cache || !*local_cache) {
```

Comment:
On Tue, 2015-01-27 at 06:03 -0800, James Bonfield wrote:
> In cram/cram_io.c:
> 
> >  	ref_path = "http://www.ebi.ac.uk:80/ena/cram/md5/%s";
> > +	if (!local_cache || !*local_cache) {
> 
> It's a stylistic thing, but grepping around I see htslib and samtools
> have many more uses of !*str than str[0] == '\0' so we may as well
> keep it as is.

Beauty (and readability!) is in the eye of the beholder. I support
James's and Heng's style, yet I am not insisting that others obey it.
Can this be mutual please?

petr



-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.